synapse/lib/cell.py

```diff
@@ -1199,6 +1202,23 @@ async def __anit__(self, dirn, conf=None, readonly=False, parent=None):
         # phase 5 - service networking
         await self.initServiceNetwork()
 
+    async def getPermDef(self, perm):
+        if self.permlook is None:
+            self.permlook = {pdef['perm']: pdef for pdef in await self.getPermDefs()}
+        return self.permlook.get(perm)
+
+    async def getPermDefs(self):
+        if self.permdefs is None:
+            self.permdefs = await self._getPermDefs()
+        return self.permdefs
+
+    async def _getPermDefs(self):
+        return ()
```

Comment:
The following permissions are used in the base Cell / CellApi implementation.

```
health
auth.self.set.passwd
auth.user.set.passwd
task.get
task.del
```

So there is a mixture of "base class perms + subclass perms" that we need to consider.

Reply:
maybe we have a `_getBasePermDefs()` and `_getPermDefs()` - both of which return tuples.

then we have the following:

```
    async def getPermDefs(self):
        if self.permdefs is None:
            self.permdefs = await self._getBaseDefs() + await self._getPermDefs()
        return self.permdefs
```
?

Reply:
The ``s_httpapi.Handler`` also uses a `impersonate` permission which is probably better moved into a base handler for the Storm APIs, since there isn't much other use for it's `_reqValidOpts()` outside of those storm cases?

Reply:
We could easily call the parent definition of this function or stuff things into a base list. The main idea was that making it a function gave us the greatest flexibility. I still have quite a number permissions to plumb in a separate PR, just wanted to get the concept plumbed.